### PR TITLE
Fixes #15755 - handle race condition on pool import

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: app/models/katello/glue/candlepin/product.rb [hunk 1]
```diff
@@ -139,13 +139,13 @@ def product_content_by_name(content_name)
...
+        ::Katello::Util::Support.active_record_retry do
+          sub = ::Katello::Subscription.where(:cp_id => subscription_id).first_or_create
+        end
         sub.import_data
         pools = ::Katello::Resources::Candlepin::Product.pools(self.organization.label, self.cp_id)
-        pools.each do |pool_json|
-          pool = ::Katello::Pool.where(:cp_id => pool_json['id']).first_or_create
-          pool.import_data
-        end
+        pools.each { |pool_json| ::Katello::Pool.import_pool(pool_json['id']) }
```
Comment (bbuckingham) [inline app/models/katello/glue/candlepin/product.rb]: Is it no longer necessary to 'import_data' at this point?

Comment (jlsherrill) [inline app/models/katello/glue/candlepin/product.rb]: it is, notice i'm calling  Pool.import_pool which handles the creation and import_data.  The code was duplicated here so i was DRYing it out a bit.

Comment (bbuckingham) [inline app/models/katello/glue/candlepin/product.rb]: Thanks!

**Comment by bbuckingham** [comment]
The code changes look good.

I was not able to reproduce the race condition that this is intended to solve; however, I was able to confirm that I can still create products and repositories as expected.

ACK; however, feel free to wait on merging if those that were encountering the race condition are in the process of testing as well.

**Comment by ehelms** [comment]
[test]

**Pull request merged**